src/frontend/js/components/CourseProductItem/index.spec.tsx

```diff
@@ -67,30 +62,40 @@ describe('CourseProductItem', () => {
     fetchMock.restore();
     JoanieEnrollmentFactory.afterGenerate((e: Enrollment) => e);
     ProductFactory.afterGenerate((p: Product) => p);
-    OrderLiteFactory.afterGenerate((o: OrderLite) => o);
+    OrderFactory.afterGenerate((o: Order) => o);
   });
 
-  const Wrapper = ({ code, children }: PropsWithChildren<{ code: string }>) => (
+  const Wrapper = ({ children }: PropsWithChildren<{}>) => (
     <IntlProvider locale="en">
-      <CourseCodeProvider code={code}>
-        <JoanieApiProvider>
-          <QueryClientProvider client={createTestQueryClient()}>{children}</QueryClientProvider>
-        </JoanieApiProvider>
-      </CourseCodeProvider>
+      <JoanieApiProvider>
+        <QueryClientProvider client={createTestQueryClient()}>{children}</QueryClientProvider>
+      </JoanieApiProvider>
     </IntlProvider>
   );
 
-  it('renders product information', () => {
+  it('renders product information', async () => {
     const product: Product = ProductFactory.generate();
+    const productDeferred = new Deferred();
+    fetchMock.get(
+      `https://joanie.test/api/v1.0/products/${product.id}/?course=00000`,
+      productDeferred.promise,
+    );
 
     render(
-      <Wrapper code="00000">
-        <CourseProductItem product={product} />
+      <Wrapper>
+        <CourseProductItem courseCode="00000" productId={product.id} />
       </Wrapper>,
     );
 
-    screen.getByRole('heading', { level: 3, name: product.title });
-    // the price shouldn't be a heading to prevent misdirection for screen reader users
+    // - A loader should be displayed while product information are fetching
+    screen.getByRole('status', { name: 'Loading product information...' });
+
+    productDeferred.resolve(product);
+
+    await waitFor(() => {
+      screen.getByRole('heading', { level: 3, name: product.title });
+    });
```

Comment:
I think you could instead use 

```suggestion
await screen.findByRole('heading', { level: 3, name: product.title });
```